File: Logging.Client/ILog.cs
using System.Collections.Generic;

namespace Logging.Client
{
    public interface ILog
    {
        void Debug(string message);

        void Debug(string title, string message);

        void Debug(string title, string message, Dictionary<string, string> tags);

        void Info(string message);

        void Info(string title, string message);

        void Info(string title, string message, Dictionary<string, string> tags);

        void Warm(string message);

        void Warm(string title, string message);

        void Warm(string title, string message, Dictionary<string, string> tags);

        void Error(string message);

        void Error(string title, string message);

        void Error(string title, string message, Dictionary<string, string> tags);
    }
}

File: client/go/loggingclient/ILog.go
package loggingclient

// ILog loggingclient对外提供的接口
type ILog interface {
	Debug(params ...interface{})

	Info(params ...interface{})

	Warm(params ...interface{})

	Error(params ...interface{})

	Debugf(format string, params ...interface{})

	Infof(format string, params ...interface{})

	Warmf(format string, params ...interface{})

	Errorf(format string, params ...interface{})

	Metric(name string, value float64, tags map[string]string)
}
